src/test/java/org/gridsuite/study/server/utils/SendInput.java

```diff
@@ -40,14 +43,32 @@ public String getName() {
     public void doAction(ServeEvent serveEvent, Admin admin, Parameters parameters) {
         Object payload = parameters.get("payload");
         String destination = parameters.get("destination").toString();
+        CountDownLatch countDownLatch = (CountDownLatch) (parameters.get("latch"));
+        List<String> paramsNotToPass = List.of("countDownLatch", "destination", "payload");
 
         MessageBuilder<?> messageBuilder = MessageBuilder.withPayload(payload);
         QueryParameter receiverParam = serveEvent.getRequest().getQueryParams().get(QUERY_PARAM_RECEIVER);
         if (receiverParam != null) {
             messageBuilder.setHeader(HEADER_RECEIVER, receiverParam.firstValue());
         }
 
-        input.send(messageBuilder.build(), destination);
+        parameters.forEach((key, value) -> {
+            if (!paramsNotToPass.contains(key)) {
+                messageBuilder.setHeader(key, value);
+            }
+        });
+
+        // Wiremock does not accept to send a request http in a post serve action
```

Comment:
I though it's because we have to wait the end of the request ?

Reply:
TL found a (known) Wiremock bug which would prevent an http request to be sent in some case when ran in PostServeAction
I forgot to ask for the link though

Reply:
I mean, I though we do all this because we need to wait the end of the request, but it's not what the comment say.
I don't see the link between this comment and the code below, it's not clear to me.